File: MangaViewer/app/src/main/java/com/king/mangaviewer/model/HistoryMangaChapterItem.java
package com.king.mangaviewer.model;

import org.joda.time.DateTime;

/**
 * Created by KinG on 12/23/2015.
 */
public class HistoryMangaChapterItem extends MangaChapterItem {
    private String mLastReadDate;
    public final static String DATE_FORMAT = "yyyy-MM-dd HH:mm:ss";

    public HistoryMangaChapterItem(MangaChapterItem chapterItem){
        super(chapterItem.getId(),chapterItem.getTitle(),chapterItem.getDescription(),chapterItem.getImagePath(),chapterItem.getUrl(),chapterItem.getMenu());
        mLastReadDate = DateTime.now().toString(DATE_FORMAT);
    }

    public String getLastReadDate(){
        return mLastReadDate;
    }

}

File: MangaViewer/app/src/main/java/com/king/mangaviewer/model/HistoryMangaChapterItem.kt
package com.king.mangaviewer.model

import com.king.mangaviewer.common.Constants.DATE_FORMAT_LONG
import com.king.mangaviewer.util.StringUtils
import org.joda.time.DateTime

/**
 * Created by KinG on 12/23/2015.
 */
class HistoryMangaChapterItem(chapterItem: MangaChapterItem,
        val lastReadPageNum: Int = 0,
        val lastReadDate: String = DateTime.now().toString(DATE_FORMAT_LONG)) :
        MangaChapterItem(chapterItem.id, chapterItem.title, chapterItem.description,
                chapterItem.imagePath, chapterItem.url, chapterItem.menu) {

}
